src/validation.js

```diff
@@ -1,10 +1,15 @@
 import { NamedNode } from "rdflib";
-import { FORM, RDF } from "./namespaces.js";
+import { FORM, RDF, SHACL } from "./namespaces.js";
 import { fieldsForForm, fieldsForSubForm } from "./fields-for-form.js";
 import { check, checkTriples } from "./constraints.js";
 import { getScope } from "./get-scope.js";
 import { triplesForScope } from "./triples-for/triples-for-scope.js";
 
+/**
+ * @typedef {import('forking-store').default} ForkingStore
```

Comment:
thinking: not sure how i feel about these... on the one hand it's nice to have type information, on the other it feels like wanting to do typescript but not being able to do it and only going half way... I know ember isn't quite ready for typescript yet so maybe this is the best we can get, but not sure how much it adds really

Reply:
This is more a documentation thing for me at the moment, not for actual typechecking.

Agreed that converting to actual TS would be better (or at least, have it check the types). I think we can easily do it here (and in the forking store) since they are regular npm packages. We just need to convince @cecemel first 😬.

Also, I don't think modern Ember has that many issues with TS? I think the last remaining hurdle is the outdated EmberData types and lack of official EmberData types. Also, check out [Glint](https://typed-ember.gitbook.io/glint/) if you haven't heard of it yet.